data/api/client-server/profile.yaml

```diff
@@ -16,48 +16,105 @@ info:
+      x-changedInMatrixVersion:
+        "1.14": Endpoint now accepts variable `keyName` parameter.
+      summary: Set a profile field for a user.
       description: |-
-        This API sets the given user's display name. You must have permission to
-        set this user's display name, e.g. you need to have their `access_token`.
-      operationId: setDisplayName
+        Set or update a profile field for a user. Must be authenticated with an
+        access token authorised to make changes. Servers may impose size limits
+        on individual fields, and the total profile must be under 64 KiB.
+
+        **Note**: Setting a field to `null` keeps the key but with a `null` value,
+        which some servers may reject. To remove a field completely, use the
+        `DELETE` endpoint instead.
+      operationId: setProfileField
       security:
         - accessTokenQuery: []
         - accessTokenBearer: []
       parameters:
         - in: path
           name: userId
-          description: The user whose display name to set.
+          description: The user whose profile field to set.
           required: true
           example: "@alice:example.com"
           schema:
             type: string
+        - in: path
+          name: keyName
+          description: The profile field key name to set. It must be either
+            `avatar_url`, `displayname`, or a custom field following the
+            [Common Namespaced Identifier Grammar](/appendices/#common-namespaced-identifier-grammar).
+          required: true
+          example: "displayname"
+          schema:
+            type: string
+            pattern: '^(avatar_url|displayname|[a-z][a-z0-9_]*(\.[a-z][a-z0-9_]*)+)$'
       requestBody:
+        description: A JSON object containing the property whose name matches
+          the `keyName` specified in the URL. See `additionalProperties` for
+          further details.
+        required: true
         content:
           application/json:
             schema:
               type: object
-              example: {
-                "displayname": "Alice Margatroid"
-              }
-              properties:
-                displayname:
-                  type: string
-                  description: The new display name for this user.
-        description: The new display name information.
-        required: true
+              minProperties: 1
+              additionalProperties:
+                description: The JSON object must include a property whose key
+                  matches the `keyName` specified in the URL. For `avatar_url`,
+                  the value must be an MXC URI string. For `displayname`, the value
+                  must be a string. For custom keys, any JSON type is allowed -
+                  servers may not validate these values, but clients should follow
+                  the format defined for that key.
+              example: { "displayname": "Alice Wonderland" }
```

Comment:
The templating stuff seems to be having a hard time rendering this:

![image](https://github.com/user-attachments/assets/e9db021e-1bee-4cb0-b4d7-f0ae34d1e79c)

@zecakeh, do you happen to have any idea why this is breaking?

Reply:
It looks like we're in [the branch where _only_ `additionalProperties` is set](https://github.com/matrix-org/matrix-spec/blob/main/layouts/partials/openapi/render-object-table.html#L83). In this case, the `description` is read from the [top level](https://github.com/matrix-org/matrix-spec/blob/main/layouts/partials/openapi/render-object-table.html#L310) rather than from `additionalProperties.description`. This is empty here though.

Consequently, moving the description one level up produces this:

<img width="853" height="244" alt="Screenshot 2025-07-30 at 12 44 43" src="https://github.com/user-attachments/assets/52d7c550-df61-4064-8782-5451ba340690" />